crankshaft-engine/src/service/runner/backend/docker.rs

```diff
@@ -35,11 +97,156 @@ impl Backend {
     /// Note that, currently, we connect [using
     /// defaults](Docker::connect_with_defaults) when attempting to connect to
     /// the Docker daemon.
-    pub fn initialize_default_with(config: Config) -> Result<Self> {
-        let client = Docker::with_defaults()
-            .context("error connecting to the Docker daemon—is it running?")?;
+    pub async fn initialize_default_with(config: Config) -> Result<Self> {
+        let client =
+            Docker::with_defaults().context("failed to connect to the local Docker daemon")?;
+
+        let info = client
+            .info()
+            .await
+            .context("failed to retrieve local Docker daemon information")?;
+
+        // Check to see if the daemon is part of an active swarm or not
+        // If the daemon is part of a swarm, but the node is not active or a manager, we
+        // can't spawn tasks
+        let swarm = if let Some(swarm) = &info.swarm {
+            match (&swarm.node_id, swarm.local_node_state) {
+                (Some(id), Some(LocalNodeState::ACTIVE)) if !id.is_empty() => {
+                    // Part of an active swarm, check to see if the node is a manager
+                    if !swarm.control_available.unwrap_or(false) {
```

Comment:
Is there a place where this default is documented? If so, perhaps we should link there?

Reply:
I'll add a link to [Docker's system information API](https://docs.docker.com/reference/api/engine/version/v1.47/#tag/System/operation/SystemInfo), which defaults it to false.